Replace GitHub markdown templates with GitHub forms

xavier2k6 opened the pull request:
Reason for this PR:
>It would be just fine if GitHub allowed us to have a "Create new Issue" form (instead of Issue template) where we would place all these prerequisites, checkboxes, etc. in a friendly way, and it would simply not allow someone to create an Issue if some of the required fields are not filled in or filled in incorrectly.

https://github.com/qbittorrent/qBittorrent/pull/12574#issuecomment-723785514

With this PR:
Updates the bug report & feature request templates to the [new form-based format](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository#creating-issue-forms).

- Delete bug_report.md
- Delete feature_request.md
- Create bug_report.yml
- Create feature_request.yml
- fix indentations and formatting (config file)

Reference:
[bug_report.yml](https://github.com/xavier2k6/qBittorrent/blob/github_forms/.github/ISSUE_TEMPLATE/bug_report.yml)
[feature_request.yml](https://github.com/xavier2k6/qBittorrent/blob/github_forms/.github/ISSUE_TEMPLATE/feature_request.yml)

Pros:
- [Different types of form elements](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-githubs-form-schema#keys)
- No more template evading/removed submissions (best effort)
- Users are more likely to fill it properly as it is easier to read
- Every required field is actually enforced
- Integrates https://github.com/qbittorrent/qBittorrent/pull/15159
- All Mandatory checklists/prerequisites in one place
- All Optional checklists/prerequisites in one place
____

**Issue form screenshots:**

<details>
<summary>What user is presented with when creating an issue</summary>

![What user is presented with when creating an issue V1](https://user-images.githubusercontent.com/42386382/132993043-192371ca-eaa5-4448-b085-f720cdeacda7.png)

</details>

<details>
<summary>What the team will see when an issue is submitted</summary>

![What the team will see when an issue is submitted V1](https://user-images.githubusercontent.com/42386382/132993034-ee067694-e724-42e7-8e19-05e2fa10c577.png)

</details>

Example:
https://github.com/xavier2k6/qBittorrent/issues/27
____

**Feature request screenshots:**

<details>
<summary>What user is presented with when creating a feature request</summary>

![What user is presented with when creating a feature request V1](https://user-images.githubusercontent.com/42386382/132993029-5d67e094-9d57-4931-9324-e29705181c19.png)

</details>

<details>
<summary>What the team will see when feature request is submitted</summary>

![What the team will see when feature request is submitted V1](https://user-images.githubusercontent.com/42386382/132993011-54502117-ef43-4a59-9ff5-1a67f71ec5cd.png)

</details>

Example:
https://github.com/xavier2k6/qBittorrent/issues/28
____

#### Testing environment
https://github.com/xavier2k6/qBittorrent/issues/new/choose

glassez commented:
@xavier2k6 
Please provide screenshots/examples of how forms and produced Issues will look.

thalieht commented:
I would:
- Re-merge OS and qBt version and now that we can have a description, i would also merge the "Qt & libtorrent-rasterbar versions." section into it as part of the description. Main motivator for these: Less scrolling!
- Remove the "Browser name and version." section. The issues where these are a factor are too few to deserve a whole section. 
- Make the checkbox and section about log/settings optional unless you want to see it blindly ticked and the section filled with "asdfasdfasdf" to bypass the requirement. As i said in the previous PR, i would make it opt-out as "Include them unless you know they are irrelevant to the issue" which is what happens anyway in practice.

>Please provide screenshots/examples of how forms and produced Issues will look.

https://github.com/xavier2k6/qBittorrent/issues/new?assignees=&labels=&template=bug_report.yml
https://github.com/xavier2k6/qBittorrent/issues/new?assignees=&labels=Feature+Request&template=feature_request.yml
I can't see "Preview" though. I guess someone will have to make a test issue for that.

thalieht commented on .github/ISSUE_TEMPLATE/bug_report.yml:
```diff
@@ -0,0 +1,99 @@
+
+  - type: textarea
+    attributes:
+      label: What issue are you experiencing?
```
The previous one was ok (What is the problem) but if you want to change it, you shouldn't address the reporter directly.

xavier2k6 replied:
What is the problem being experienced?

xavier2k6 replied:
Used original (What is the problem?)

thalieht commented on .github/ISSUE_TEMPLATE/bug_report.yml:
```diff
@@ -0,0 +1,99 @@
+      label: Additional context.
+      description: Add screenshots etc. (Anything that will give us more context about the issue you are encountering!)
+    validations:
+      required: true
```
false

xavier2k6 replied:
changed

thalieht commented on .github/ISSUE_TEMPLATE/bug_report.yml:
```diff
@@ -0,0 +1,99 @@
+  - type: markdown
+    attributes:
+     value: |
+             ### Thanks for taking the time to fill out this report! (qBittorrent Bug Handler Team)
```
I don't know if this will be visible in the posted issue. If it is, i would remove it completely but if it isn't (and you insist on keeping it) i would make it more formal: 
"Thank you for taking the time to fill out this report."

sakkamade replied:
> I don't know if this will be visible in the posted issue

No, it is not, not the first one at the least.
The `###` is superfluous.

> "Thank you for taking the time to fill out this report."

:+1:

xavier2k6 replied:
It's not visible & I don't insist on keeping any of it - just added it in as other repos use similar.
I will remove.

xavier2k6 replied:
it was removed.

xavier2k6 commented:
>Please provide screenshots/examples of how forms and produced Issues will look.

example of both forms can be utilized/created from my repo:
https://github.com/xavier2k6/qBittorrent/issues/new/choose

(test away creating issues/feature requests)

>I can't see "Preview" though. I guess someone will have to make a test issue for that.

There's no actual "Preview" like we know/use to.

The "Preview" appears in the (text boxes) ->"What issue are you experiencing?" etc.

@thalieht No bother, will address your comments.

sakkamade commented on .github/ISSUE_TEMPLATE/bug_report.yml:
```diff
@@ -0,0 +1,99 @@
+      label: Qt & libtorrent-rasterbar versions.
+      description: (Linux OS Only)
+    validations:
+      required: true
```
Should be `false` here.

sakkamade replied:
And in `Browser name and version`

sakkamade replied:
Otherwise change add the `value` property the with `(Linux OS Only)`

xavier2k6 replied:
applied different approach

sakkamade commented:
> example of both forms can be utilized/created from my repo:
https://github.com/xavier2k6/qBittorrent/issues/new/choose

Even so, the screenshots would not be redundant, I think.

thalieht commented on .github/ISSUE_TEMPLATE/feature_request.yml:
```diff
@@ -0,0 +1,48 @@
+
+  - type: textarea
+    attributes:
+      label: What is your Suggestion/Feature Request?
```
The titles from the previous template were adequate IMO: "Suggestion" "Use case" and "Extra info/examples/attachments".

xavier2k6 replied:
will address

thalieht commented on .github/ISSUE_TEMPLATE/bug_report.yml:
```diff
@@ -0,0 +1,99 @@
+          required: true
+        - label: This report is **not a support request or question**, both of which are better suited for either the [discussions section](https://github.com/qbittorrent/qBittorrent/discussions), [forum](https://qbforums.shiki.hu/), or [subreddit](https://www.reddit.com/r/qBittorrent/). The [wiki](https://github.com/qbittorrent/qBittorrent/wiki) did not contain a suitable solution either.
+          required: true
+        - label: I have taken note of the **(ADVISORY)** above & **added the settings file and relevant log(s)** in the **Logs & Settings/Config files** section at the bottom of the report.
```
You forgot to re-add this part:
See [this wiki page](https://github.com/qbittorrent/qBittorrent/wiki/Frequently-Asked-Questions#Where_does_qBittorrent_save_its_settings) if you're not sure where to find them.

xavier2k6 replied:
will address

Chocobo1 commented on .github/ISSUE_TEMPLATE/bug_report.yml:
```diff
@@ -0,0 +1,99 @@
+
+  - type: input
+    attributes:
+      label: Operating System & build/version.
```
I would take off the period of the label/title, especially when the label is followed by an input box.
The period looks ok in description though.

sakkamade commented:
Perhaps the `<details>` element can be added to  `Log & Settings file(s).` section?

```html
<details><summary>Log & Settings file(s)</summary>

(type here)

</details>
```
<details><summary>Log & Settings file(s)</summary>

(type here)

</details>

Chocobo1 commented on .github/ISSUE_TEMPLATE/bug_report.yml:
```diff
@@ -0,0 +1,99 @@
+  - type: input
+    attributes:
+      label: Qt & libtorrent-rasterbar versions.
+      description: (Linux OS Only)
```
This shouldn't be "Linux OS Only".

xavier2k6 replied:
>If on Linux, libtorrent-rasterbar and Qt versions

This is what we use currently?

If I merge some of the `input` fields......

Will -> qBittorrent/Qt/libtorrent-rasterbar versions and Operating system. suffice?

xavier2k6 replied:
>Will -> qBittorrent/Qt/libtorrent-rasterbar versions and Operating system. suffice?

example of `textarea` with a placeholder:
![textarea with placeholder](https://user-images.githubusercontent.com/42386382/129755037-9c3bffdd-b510-4dbe-aaec-b3ceb812dd85.png)

Chocobo1 commented on .github/ISSUE_TEMPLATE/bug_report.yml:
```diff
@@ -0,0 +1,99 @@
+    validations:
+      required: true
+
+  - type: input
```
A single line input box for 2 version strings? IMO the result would look bad, you should use `textarea` instead.
Please check other input fields.

glassez replied:
>A single line input box for 2 version strings? IMO the result would look bad, you should use `textarea` instead.

Why not separate input boxes?

Chocobo1 commented on .github/ISSUE_TEMPLATE/bug_report.yml:
```diff
@@ -0,0 +1,99 @@
+      label: Browser name and version.
+      description: (If WebUI related Only)
+    validations:
+      required: true
```
If it is "WebUI related Only", then this should be `false`.

xavier2k6 replied:
This `input` has been previously requested to be completely removed....

Chocobo1 commented on .github/ISSUE_TEMPLATE/bug_report.yml:
```diff
@@ -0,0 +1,99 @@
+    attributes:
+     value: |
+             #### ADVISORY
+             We **DO NOT** support any **End-Of-Life** versions `<=4.1.x` or **3rd party/forked versions** e.g. `portableapps`/`Enhanced Edition`etc.
```
Too many bold text and full uppercase words makes it hard to read (it looks like written by an edgy kid and put me off), please refrain from using it or use it scarcely.

The same applies to other labels.

Chocobo1 replied:
About the EOL versions, I would state it as:
"We do not support any versions older than the current release series"
This is to avoid mentioning exact numbers which calls for regular maintenance for it.

xavier2k6 replied:
>or **3rd party/forked versions** e.g. `portableapps`/`Enhanced Edition`etc.

Any issues with this? (un-bolded)

xavier2k6 replied:
>Too many bold text and full uppercase words makes it hard to read (it looks like written by an edgy kid and put me off), please refrain from using it or use it scarcely.

The same applies to other labels.

Most of the bold text words are just a copy&paste of our current template

Chocobo1 commented on .github/ISSUE_TEMPLATE/bug_report.yml:
```diff
@@ -0,0 +1,99 @@
+  - type: input
+    attributes:
+      label: Browser name and version.
+      description: (If WebUI related Only)
```
No need for parentheses IMO.

xavier2k6 replied:
this section is being removed

Chocobo1 commented on .github/ISSUE_TEMPLATE/bug_report.yml:
```diff
@@ -0,0 +1,99 @@
+      label: Steps to Reproduce.
+      description: Please walk us through it with a clear and concise description.
+      placeholder: |
+        1. [First Step]
```
No need for brackets

xavier2k6 replied:
will address

Chocobo1 commented on .github/ISSUE_TEMPLATE/bug_report.yml:
```diff
@@ -0,0 +1,99 @@
+      label: Log & Settings file(s).
+      description: Please add the settings file and at least the most recent log. See [this wiki page](https://github.com/qbittorrent/qBittorrent/wiki/Frequently-Asked-Questions#Where_does_qBittorrent_save_its_settings) if you're not sure where to find them.
+    validations:
+      required: true
```
I would prefer it is `false`.

xavier2k6 replied:
ok

Chocobo1 commented on .github/ISSUE_TEMPLATE/bug_report.yml:
```diff
@@ -0,0 +1,99 @@
+  - type: markdown
+    attributes:
+     value: |
+             #### ADVISORY
```
Too much indent. Please check the indentation of the file thoroughly.

xavier2k6 replied:
ok

xavier2k6 commented:
I will address all issues & add screenshots etc.

Will convert to draft for now.

glassez commented:
My general opinion is as follows:
The biggest and undoubted advantage of these changes is that it will no longer produce this extra annoying text in the preamble of the Issues. So I suggest not to hung up on some little (subjective) things now, but simply eliminate obvious shortcomings (such as extra dots at the end of input field labels) and merge it. Fine-tuning can be done later in one of the following PRs.

xavier2k6 commented:
OP updated with screenshot & issue example ref https://github.com/qbittorrent/qBittorrent/pull/15330#issue-713627701

"feature request" changes are a WIP..........time needed.

PriitUring commented on .github/ISSUE_TEMPLATE/bug_report.yml:
```diff
@@ -4,96 +4,75 @@ body:
+        required: true
+      - label: This report is **not a support request or question**, both of which are better suited for either the [discussions section](https://github.com/qbittorrent/qBittorrent/discussions), [forum](https://qbforums.shiki.hu/), or [subreddit](https://www.reddit.com/r/qBittorrent/). The [wiki](https://github.com/qbittorrent/qBittorrent/wiki) did not contain a suitable solution either.
+        required: true
+      - label: I have taken note of the **(ADVISORY)** & if releavant to issue or if asked by a **team member** will add the settings file and log(s) in the **Logs & settings file(s)** section at the bottom of the report. See [this wiki page](https://github.com/qbittorrent/qBittorrent/wiki/Frequently-Asked-Questions#Where_does_qBittorrent_save_its_settings) if you're not sure where to find them.
```
Typo: word "releavant" should be replaced with "relevant"?

PriitUring commented:
Typo: word "releavant" should be written as "relevant" in the last checkmark sentence in the Issue"

PriitUring commented on .github/ISSUE_TEMPLATE/bug_report.yml:
```diff
@@ -4,96 +4,75 @@ body:
+        required: true
+      - label: This report is **not a support request or question**, both of which are better suited for either the [discussions section](https://github.com/qbittorrent/qBittorrent/discussions), [forum](https://qbforums.shiki.hu/), or [subreddit](https://www.reddit.com/r/qBittorrent/). The [wiki](https://github.com/qbittorrent/qBittorrent/wiki) did not contain a suitable solution either.
+        required: true
+      - label: I have taken note of the **(ADVISORY)** & if releavant to issue or if asked by a **team member** will add the settings file and log(s) in the **Logs & settings file(s)** section at the bottom of the report. See [this wiki page](https://github.com/qbittorrent/qBittorrent/wiki/Frequently-Asked-Questions#Where_does_qBittorrent_save_its_settings) if you're not sure where to find them.
```
Typo: word "releavant" should be replaced with "relevant".

xavier2k6 replied:
Fixed.

sakkamade commented on .github/ISSUE_TEMPLATE/bug_report.yml:
```diff
@@ -0,0 +1,78 @@
+
+  - type: textarea
+    attributes:
+      label: What is the problem being experienced?
```
`What problem is being experienced?`

sakkamade commented on .github/ISSUE_TEMPLATE/bug_report.yml:
```diff
@@ -0,0 +1,78 @@
+  - type: textarea
+    attributes:
+      label: What is the expected behavior?
+      description: Please give a clear and concise description of what you expected to happen.
```
User should not be addressed, right?

`(...) description of expected behaviour.`

thalieht replied:
>User should not be addressed, right?

I think it's ok in the description because only the reporter can see this.

xavier2k6 replied:
section removed.

sakkamade commented on .github/ISSUE_TEMPLATE/bug_report.yml:
```diff
@@ -0,0 +1,78 @@
+  - type: textarea
+    attributes:
+      label: What is the problem being experienced?
+      description: Please add "crash report" (if encountered) or give a clear and concise description of the bug/issue.
```
`Please add the "crash report" (if encountered) or provide a clear and concise description of the problem.`

sakkamade commented on .github/ISSUE_TEMPLATE/bug_report.yml:
```diff
@@ -0,0 +1,78 @@
+  - type: textarea
+    attributes:
+      label: Additional context.
+      description: Add screenshots etc. (Anything that will give us more context about the issue you are encountering!)
```
Vocabulary should be constant I think, i.e. if there has already been used the word "problem", it should be kept being used.
```
A screenshots, screencasts, etc. 
(Anything that will provide more detailed description of the problem.)
```

sakkamade commented on .github/ISSUE_TEMPLATE/bug_report.yml:
```diff
@@ -0,0 +1,78 @@
+  - type: textarea
+    attributes:
+      label: Steps to reproduce.
+      description: Please walk us through it with a clear and concise description.
```
`Please provide a reliable steps to reproduce the problem.`

sakkamade commented on .github/ISSUE_TEMPLATE/bug_report.yml:
```diff
@@ -0,0 +1,78 @@
+  - type: textarea
+    attributes:
+      label: Log & settings file(s).
+      description: Please add the settings file and at least the most recent log. See [this wiki page](https://github.com/qbittorrent/qBittorrent/wiki/Frequently-Asked-Questions#Where_does_qBittorrent_save_its_settings) if you're not sure where to find them.
```
`(...) See ["Where does qBittorrent save its settings"](link).`

sakkamade commented on .github/ISSUE_TEMPLATE/bug_report.yml:
```diff
@@ -0,0 +1,78 @@
+
+  - type: textarea
+    attributes:
+      label: Steps to reproduce.
```
Periods are still left. in several labels.

thalieht commented on .github/ISSUE_TEMPLATE/bug_report.yml:
```diff
@@ -0,0 +1,78 @@
+        required: true
+      - label: This report is **not a support request or question**, both of which are better suited for either the [discussions section](https://github.com/qbittorrent/qBittorrent/discussions), [forum](https://qbforums.shiki.hu/), or [subreddit](https://www.reddit.com/r/qBittorrent/). The [wiki](https://github.com/qbittorrent/qBittorrent/wiki) did not contain a suitable solution either.
+        required: true
+      - label: I have taken note of the **(ADVISORY)** & if relevant to issue or if asked by a **team member** will add the settings file and log(s) in the **Logs & settings file(s)** section at the bottom of the report. See [this wiki page](https://github.com/qbittorrent/qBittorrent/wiki/Frequently-Asked-Questions#Where_does_qBittorrent_save_its_settings) if you're not sure where to find them.
```
The reporter should know which parts are required and which are optional, like the 3rd checkbox (optional). Same goes for the fields below, as a placeholder maybe?

thalieht commented on .github/ISSUE_TEMPLATE/bug_report.yml:
```diff
@@ -0,0 +1,78 @@
+
+  - type: textarea
+    attributes:
+      label: Operating System & qBittorrent / Qt / libtorrent-rasterbar versions
```
This section bugs me. You are forcing the reporter to unconditionally mention Qt and libtorrent versions but if they're on Windows, there's a 99% chance it's whatever the official release uses. So i would move Qt/libt to description:
"If (you are) on linux, mention your libtorrent-rasterbar and Qt version."

I would prefer it if there was no placeholder text in this section so that the reporter can format it however they want but i don't insist on this as it is purely personal preference.

thalieht replied:
"Operating System & qBittorrent"
I would also re-arrange these so qBittorrent comes first.

Chocobo1 replied:
I suppose this could just become `qBittorrent & operating system versions` and mention the version requirements in the description:
```
- type: textarea
  attributes:
    label: qBittorrent & operating system versions
    description: |
      Qt and libtorrent-rasterbar versions are required when: 1. You are using linux. 2. You are not using an official build downloaded from our website.
      Example: 
        - qBittorrent: 4.3.7 x64
        - Operating system: Windows 10 Pro 21H1/2009 x64
        - Qt: 5.15.2
        - libtorrent-rasterbar: 1.2.14
    value: |
      - qBittorrent:
      - Operating system:
      - Qt:
      - libtorrent-rasterbar:
```

Some notes:
* I would prefer prefill the textarea to have consistent formatting.
* Theoretically 32, 64-Bit shouldn't make a difference and since we are slowly moving away from 32-bit, I suggest omitting it in description.

thalieht commented on .github/ISSUE_TEMPLATE/bug_report.yml:
```diff
@@ -0,0 +1,78 @@
+      label: What is the expected behavior?
+      description: Please give a clear and concise description of what you expected to happen.
+    validations:
+     required: true
```
Is this section really required to be filled? I would even question its usefulness because it's (almost?) always the opposite of the problem.

xavier2k6 replied:
section was removed.

thalieht commented on .github/ISSUE_TEMPLATE/feature_request.yml:
```diff
@@ -0,0 +1,43 @@
+  - type: textarea
+    attributes:
+      label: Suggestion
+      description: Please give a clear and a concise description.
```
Typo: extra "a" before concise.

Personal opinion: "Suggestion" and "Use case" don't need a description.

Chocobo1 replied:
@xavier2k6 @thalieht 
Is this comment still valid?

xavier2k6 replied:
>Is this comment still valid?

Good catch!

Looks like I overlooked this, I will remove the descriptions completely instead of just the typo unless anybody else feels they should stay?! (I have no strong opinion either way)

xavier2k6 replied:
descriptions removed.

Chocobo1 commented on .github/ISSUE_TEMPLATE/bug_report.yml:
```diff
@@ -0,0 +1,78 @@
+      #### ADVISORY
+       "We do not support any versions older than the current release series" or 3rd party/forked versions e.g. `portableapps`/`Enhanced Edition`etc.
+       It is the responsibility of the user to **redact any sensitive information** in logs/config files.
+  - type: checkboxes
```
Add an empty line above please.

xavier2k6 replied:
>I would prefer prefill the textarea to have consistent formatting.

Just a note on this, If it was prefilled - it would invalidate the true/false requirement.

We could be left with just:
      qBittorrent:
      Operating system:
      Qt:
      libtorrent-rasterbar:

I added a comment in description about how we would like it formatted & added them as a placeholder instead.

Chocobo1 commented on .github/ISSUE_TEMPLATE/bug_report.yml:
```diff
@@ -0,0 +1,78 @@
+     value: |
+      #### ADVISORY
+       "We do not support any versions older than the current release series" or 3rd party/forked versions e.g. `portableapps`/`Enhanced Edition`etc.
+       It is the responsibility of the user to **redact any sensitive information** in logs/config files.
```
The above 2 lines should have the same indent level as `#### ADVISORY`.

xavier2k6 commented:
will address issues/comments etc in a few days, haven't time right now.

ArcticGems commented:
The new  screenshots looks great, so hopefully it gets merged soon. 👍

xavier2k6 commented on .github/ISSUE_TEMPLATE/bug_report.yml:
```diff
@@ -0,0 +1,78 @@
+body:
+  - type: markdown
+    attributes:
+     value: |
```
Thank you for the tool

xavier2k6 commented:
> The new screenshots looks great, so hopefully it gets merged soon. 👍

Thanks, I've re-arranged things/re-worded etc. - So hopefully this will suffice for now.

OP Updated with New Screenshots/Example links etc.

thalieht commented on .github/ISSUE_TEMPLATE/bug_report.yml:
```diff
@@ -0,0 +1,95 @@
+
+  - type: checkboxes
+    attributes:
+      label: Optional prerequisites
```
Honestly i'm not a fan of the creation of a whole section for what should be 1 checkbox (more scrolling...). Who's ever going to check more than one of the CI test builds anyway? And if you say it's for our convenience, it's totally not. We'll know which CI build they tested by the user's OS. In the end, i doubt we even care which OS they tested it on.

ArcticGems replied:
Hmm, that makes sense. Maybe 'Optional prerequisites' section should be removed.

xavier2k6 replied:
>Honestly i'm not a fan of the creation of a whole section for what should be 1 checkbox (more scrolling...). Who's ever going to check more than one of the CI test builds anyway? And if you say it's for our convenience, it's totally not. We'll know which CI build they tested by the user's OS. In the end, i doubt we even care which OS they tested it on.

I only included it in to it's own section as we had already asked about the CI builds, since we now have segregated OS builds & to separate mandatory/optional.

ref.:
> - [ ] (optional but recommended) I have verified that the **issue is not fixed and is reproducible** in the **[latest CI build](https://github.com/qbittorrent/qBittorrent/actions/workflows/ci.yaml?query=branch%3Amaster+event%3Apush)**
____

I agree with general consensus to remove 'Optional prerequisites' section & I will include the CI builds with the "official qBittorrent version"

Example:
>I have verified that the issue is not fixed and is reproducible in the **[latest official qBittorrent version](https://www.qbittorrent.org/download.php)** or in any of the relevant **[latest CI macOS build](https://github.com/qbittorrent/qBittorrent/actions/workflows/ci_macos.yaml?query=branch%3Amaster+event%3Apush)**/**[latest CI Ubuntu build](https://github.com/qbittorrent/qBittorrent/actions/workflows/ci_ubuntu.yaml?query=branch%3Amaster+event%3Apush)**/**[latest CI Windows build](https://github.com/qbittorrent/qBittorrent/actions/workflows/ci_windows.yaml?query=branch%3Amaster+event%3Apush).**

@thalieht Is this ok with you?

ArcticGems replied:
> I have verified that the issue is not fixed and is reproducible in the **[latest official qBittorrent version](https://www.qbittorrent.org/download.php)** or in any of the relevant **[latest CI macOS build](https://github.com/qbittorrent/qBittorrent/actions/workflows/ci_macos.yaml?query=branch%3Amaster+event%3Apush)**/**[latest CI Ubuntu build](https://github.com/qbittorrent/qBittorrent/actions/workflows/ci_ubuntu.yaml?query=branch%3Amaster+event%3Apush)**/**[latest CI Windows build](https://github.com/qbittorrent/qBittorrent/actions/workflows/ci_windows.yaml?query=branch%3Amaster+event%3Apush).**
> 

I think the last part is a bit of a mouthful. How about this?? =

> 
> I have verified that the issue is not fixed and is reproducible in the **[latest official qBittorrent version](https://www.qbittorrent.org/download.php)** or in any of the relevant latest CI builds  (**[macOS build](https://github.com/qbittorrent/qBittorrent/actions/workflows/ci_macos.yaml?query=branch%3Amaster+event%3Apush)**/**[Ubuntu build](https://github.com/qbittorrent/qBittorrent/actions/workflows/ci_ubuntu.yaml?query=branch%3Amaster+event%3Apush)**/**[Windows build](https://github.com/qbittorrent/qBittorrent/actions/workflows/ci_windows.yaml?query=branch%3Amaster+event%3Apush)**).

xavier2k6 replied:
>I think the last part is a bit of a mouthful. How about this?? =

> 
> I have verified that the issue is not fixed and is reproducible in the **[latest official qBittorrent version](https://www.qbittorrent.org/download.php)** or in any of the relevant latest CI builds  (**[macOS build](https://github.com/qbittorrent/qBittorrent/actions/workflows/ci_macos.yaml?query=branch%3Amaster+event%3Apush)**/**[Ubuntu build](https://github.com/qbittorrent/qBittorrent/actions/workflows/ci_ubuntu.yaml?query=branch%3Amaster+event%3Apush)**/**[Windows build](https://github.com/qbittorrent/qBittorrent/actions/workflows/ci_windows.yaml?query=branch%3Amaster+event%3Apush)**).

Ya, I'll go with this unless there's any objections from others.

Thanks @ArcticGems

thalieht replied:
>& to separate mandatory/optional.

There was nothing wrong with them being mixed since the optional one was denoted as such (maybe it's just me).

>& I will include the CI builds with the "official qBittorrent version"
>
>Example:
>
>>    I have verified that the issue is not fixed and is reproducible in the latest official qBittorrent version or in any of the relevant latest CI macOS build/latest CI Ubuntu build/latest CI Windows build.

That would make testing the CI builds mandatory. Are we ok with that? I think it's too much but no strong opinion from me.
If you go with that here's a shorter version (with spaces around the slashes to make it clear they are separate links):
>I have verified that the issue is not fixed and is reproducible in the latest official qBittorrent version or in the latest CI build 
(macOS / Ubuntu / Windows).

>

ArcticGems replied:
Maybe we should go back to the old one (but updated) =
> 
> * [ ]  (optional but recommended) I have verified that the issue is not fixed and is reproducible in the relevant latest CI build  (**[macOS build](https://github.com/qbittorrent/qBittorrent/actions/workflows/ci_macos.yaml?query=branch%3Amaster+event%3Apush)**/**[Ubuntu build](https://github.com/qbittorrent/qBittorrent/actions/workflows/ci_ubuntu.yaml?query=branch%3Amaster+event%3Apush)**/**[Windows build](https://github.com/qbittorrent/qBittorrent/actions/workflows/ci_windows.yaml?query=branch%3Amaster+event%3Apush)**).

I chose to keep "build" (thalieht chosed to remove it) because I feel it's more clear.

xavier2k6 replied:
>There was nothing wrong with them being mixed since the optional one was denoted as such (maybe it's just me).

I don't have any strong opinion on it either

>Maybe we should go back to the old one (but updated) =

What about this? (marked as not needing validation, of course!)

> * [ ]  (optional but recommended) I have verified that the issue is not fixed and is reproducible in the latest CI (**[macOS](https://github.com/qbittorrent/qBittorrent/actions/workflows/ci_macos.yaml?query=branch%3Amaster+event%3Apush)** / **[Ubuntu](https://github.com/qbittorrent/qBittorrent/actions/workflows/ci_ubuntu.yaml?query=branch%3Amaster+event%3Apush)** / **[Windows](https://github.com/qbittorrent/qBittorrent/actions/workflows/ci_windows.yaml?query=branch%3Amaster+event%3Apush)**) "master" builds.

ArcticGems replied:
> 
> What about this? (marked as not needing validation, of course!)
> 
> > * [ ]   (optional but recommended) I have verified that the issue is not fixed and is reproducible in the latest CI (**[macOS](https://github.com/qbittorrent/qBittorrent/actions/workflows/ci_macos.yaml?query=branch%3Amaster+event%3Apush)** / **[Ubuntu](https://github.com/qbittorrent/qBittorrent/actions/workflows/ci_ubuntu.yaml?query=branch%3Amaster+event%3Apush)** / **[Windows](https://github.com/qbittorrent/qBittorrent/actions/workflows/ci_windows.yaml?query=branch%3Amaster+event%3Apush)**) "master" builds.

It's also good 👍  

Let's see what the others think.

xavier2k6 replied:
I will also change the title from "Mandatory prerequisites" to just "Prerequisites" if we are going to go back & include the (optional but recommended) part.

ArcticGems replied:
> 
> 
> I will also change the title from "Mandatory prerequisites" to just "Prerequisites" if we are going to go back & include the (optional but recommended) part.

Agreed 👍

glassez replied:
>What about one of these?

IMO, "master" has no meaning for regular users.

xavier2k6 replied:
> IMO, "master" has no meaning for regular users.

Will omit.

xavier2k6 replied:
addressed.

thalieht reviewed (commented):
LGTM except for a couple of comments

ArcticGems commented on .github/ISSUE_TEMPLATE/bug_report.yml:
```diff
@@ -0,0 +1,95 @@
+          required: true
+        - label: I understand that it is the responsibility of the user to **redact any sensitive information** in logs/config files.
+          required: true
+        - label: I understand that if relevant to issue/when asked, I must provide the **qBittorrent.ini**, **qBittorrent.log** and **watched_folders.json** file(s) (if using "Watched Folders" feature). See **[Where does qBittorrent save its settings?](https://github.com/qbittorrent/qBittorrent/wiki/Frequently-Asked-Questions#Where_does_qBittorrent_save_its_settings)**
```
Maybe=
qBittorrent settings (.ini/.conf/.idk)

xavier2k6 replied:
>Why split the above checkbox from this one? Seems superfluous.

Will integrate.

>I would change "qBittorrent.ini" to "qBittorrent settings|config|preferences file" (pick one) because it's not ".ini" in linux, it's ".conf" (maybe mac too idk).

I will choose **qBittorrent preferences** as that's what's used if you follow [Where does qBittorrent save its settings?](https://github.com/qbittorrent/qBittorrent/wiki/Frequently-Asked-Questions#Where_does_qBittorrent_save_its_settings)

Will also change description under Log(s) & settings file(s) to reflect same.

Chocobo1 replied:
Maybe just me, if it is really necessary, I would use code blocks instead of bold text for "`qBittorrent preferences file`/`qBittorrent.log` and `watched_folders.json`"

xavier2k6 replied:
It's not 100% necessary but It's a clearer visual (to me anyway) of what files are required, code blocks work too.

I can omit any kind of formatting on them if you prefer?

Chocobo1 replied:
>I can omit any kind of formatting/highlighting on them if you prefer?

That will work for me too.

xavier2k6 replied:
all issues addressed I believe.

glassez commented:
Are there any problems with completing this PR? Why is it still a Draft?

I would still merge it ASAP, and leave fine tuning for subsequent PRs.

glassez reviewed (dismissed):
Without going into details...

xavier2k6 commented:
>Are there any problems with completing this PR? Why is it still a Draft?

I've only a few final comments to address, which will be taken care of tonight/tomorrow.

xavier2k6 commented:
@Chocobo1 @thalieht Hopefully I've addressed all of your comments/concerns now.

@glassez Apologies but your approval got kicked when I pushed changes.

[OP updated](https://github.com/qbittorrent/qBittorrent/pull/15330#issue-713627701) with new screenshots & links to example issue/feature request

thalieht commented on .github/ISSUE_TEMPLATE/bug_report.yml:
```diff
@@ -0,0 +1,84 @@
+      description: |
+        Add these files: qBittorrent preferences file, qBittorrent.log & watched_folders.json (if using "Watched Folders" feature).
+        See **[Where does qBittorrent save its settings?](https://github.com/qbittorrent/qBittorrent/wiki/Frequently-Asked-Questions#Where_does_qBittorrent_save_its_settings)**
+        #### Note: It's users responsibility to redact any sensitive information
```
Very last comment from me (same for the feature request form):
```suggestion
        #### Note: It's the user's responsibility to redact any sensitive information
```

xavier2k6 replied:
>Very last comment from me (same for the feature request form):

It's not in feature request section?!

>#### Note: It's the user's responsibility to redact any sensitive information

will change

xavier2k6 replied:
@thalieht done (bug_report)

thalieht replied:
>It's not in feature request section?!

Oops, don't know what i was looking at, sorry.

ArcticGems replied:
Just a nitpick (you don't have to do it).

Maybe the 'Extra info/examples/attachments' section in feature_request.yml should also have =

Note: It's the user's responsibility to redact any sensitive information

xavier2k6 replied:
@ArcticGems IMO I don't think it's really required as more than likely it'll be screenshots, links or a wall of text of what the user wants......... but of course that doesn't mean I won't add it or am objecting it, if others also think it's a good idea to do so - it can even be added in the future if really required.

glassez commented:
@xavier2k6 
Oops... Only now I realized the fact that it is a little not what I expected from it 😒
When I initially proposed using some forms to create Issues, I mostly wanted that the data that makes sense only at the time of creating the Issue does not fall into its contents (I mean all these Prerequisites checkboxes). Now I see (in the screenshots above) that all this "garbage" is still here. Question - does "GitHub Forms" really not allow you to do as I would like? Or did you just miss this aspect?

xavier2k6 commented:
>(I mean all these Prerequisites checkboxes). Now I see (in the screenshots above) that all this "garbage" is still here. Question - does "GitHub Forms" really not allow you to do as I would like?

All Prerequisites checkboxes can be removed (It's not a requirement in order to implement the github forms.)

>Or did you just miss this aspect?

It wasn't clear to me that you didn't fully want them. (crossed wires lol)

How about these?

What the user would see:
![new issue template](https://user-images.githubusercontent.com/42386382/133437582-054f96a3-ef30-444c-98a5-42f9b55e934f.png)

What we would see:
![issue what we see](https://user-images.githubusercontent.com/42386382/133437662-e8bf5331-a8e7-4d38-b9af-f71901165569.png)

Reference issue:
https://github.com/xavier2k6/qBittorrent/issues/29

glassez commented:
@xavier2k6 
I would want the following (if it's allowed):
1. The user sees all these Prerequisites and is required to accept them in order to create an Issue (otherwise it should be impossible).
2. When the Issue is created, it should not contain Prerequisites (the fact that the Issue is created implies that Prerequisites are accepted).

xavier2k6 commented:
>The user sees all these Prerequisites and is required to accept them in order to create an Issue.

That's basically what we have now.

>When the Issue is created, it should not contain Prerequisites

Unfortunately for now, (github forms is still in beta)  `markdown` is not rendered in created issue -> that's why we don't see the **ADVISORY etc.

We can:
1. Remove them completely, user will still have to fill in "required" text area fields etc. before being able to submit an issue.
2. Leave as-is (until `markdown` in issue body becomes available & we could use `collapsible markdown`??)
3. Instead of having them as checkboxes.......just display e.g. Please Read: xyz under `markdown` which then wouldn't be displayed in created issue.

glassez commented:
@xavier2k6 
**Option 3** looks like an acceptable compromise (if the Prerequisites are formatted as a list for better readability). 

@Chocobo1, @thalieht
What do you think about this?

xavier2k6 commented:
Example listing:
![prerequisites list](https://user-images.githubusercontent.com/42386382/133480123-401816c2-9586-49aa-b898-6a46e8c4c470.png)

thalieht commented:
>Option 3 looks like an acceptable compromise (if the Prerequisites are formatted as a list for better readability).
>
>@Chocobo1, @thalieht
>What do you think about this?

I'm fine with all of those options.

Chocobo1 commented:
>I'm fine with all of those options.

me too.

xavier2k6 commented:
>>I'm fine with all of those options.

>me too

Option 3, it is then - I'll remove the `checkboxes` from both `bug_report` / `feature_request` & reword/list(s) like https://github.com/qbittorrent/qBittorrent/pull/15330#issuecomment-920217297

(We can add back the `checkboxes` at a later stage in another PR, if we deem them necessary etc. etc.)

xavier2k6 commented:
Is everyone ok with the wording??

Regarding this part ->[Check the **[frequent/common issues list](https://github.com/qbittorrent/qBittorrent/projects/2).**] in the "feature request", is this appropriate here or should it be omitted?? (It's basically a remnance/copy&paste from the original `feature_request.md` file.

**Issue:** (User will see this beforehand - won't show in submitted body)
___
![issue rewording](https://user-images.githubusercontent.com/42386382/133609238-31a57672-2c61-4337-9728-e96bfd2fe583.png)

**Feature request:** (User will see this beforehand - won't show in submitted body)
___
![feature request rewording](https://user-images.githubusercontent.com/42386382/133609256-d65a55c8-be80-4bcd-ac88-2d74f9f2ae5b.png)

Chocobo1 commented:
@xavier2k6 
Thank you very much for getting it done!

xavier2k6 commented:
>Thank you very much for getting it done!

No problem, Looks like it's doing the job - we have a good detailed/constructive "first issue!" ref. #15476

dark-swordsman commented:
Glad to be the first one! Lol. I wasn't sure if I provided enough info honestly.

Loved the format and it was easy to follow.
